src/components/Portal/Portal.tsx

```diff
@@ -1,23 +1,20 @@
-import React, {useEffect, useMemo} from 'react'
+import React, { useEffect, useMemo } from 'react'
 import ReactDOM from 'react-dom'
 import style from './Portal.module.scss'
-const modalRootElement = document.getElementById('react-modals') as HTMLElement;
+const modalRootElement = document.getElementById('react-modals') as HTMLElement
 
-interface portalProps{
+interface portalProps {
   children?: React.ReactNode
 }
- const Portal = ({children}:portalProps) =>{
-   const  elem = useMemo(() => document.createElement('div'), [])
-  useEffect(()=>{
-    modalRootElement?.appendChild(elem);
-    return()=>{
-      modalRootElement?.removeChild(elem);
+const Portal = ({ children }: portalProps) => {
+  const elem = useMemo(() => document.createElement('div'), [])
+  useEffect(() => {
+    modalRootElement?.appendChild(elem)
+    return () => {
+      modalRootElement?.removeChild(elem)
```

Comment:
you should not work with dom elements like this, appending or removing elements is React's work. Consider conditional rendering using React itself.